src/lib/autocomplete/autocomplete-trigger.ts

```diff
@@ -129,27 +137,56 @@ export class MatAutocompleteTrigger implements ControlValueAccessor, OnDestroy {
   /** Stream of escape keyboard events. */
   private _escapeEventStream = new Subject<void>();
 
+  /** Emits once and only once on the next onStable event. */
+  private readonly _nextStable: Observable<void>;
+
+  /**  The current model value. */
+  private _value: any;
+
   /** View -> model callback called when value changes */
   _onChange: (value: any) => void = () => {};
 
   /** View -> model callback called when autocomplete has been touched */
   _onTouched = () => {};
 
   /* The autocomplete panel to be attached to this trigger. */
-  @Input('matAutocomplete') autocomplete: MatAutocomplete;
+  @Input('matAutocomplete')
+  set _autocomplete(value: MatAutocomplete) {
+    this.autocomplete = value;
+    if (this._displayWithSubscription) {
+      this._displayWithSubscription.unsubscribe();
+    }
+    if (this.autocomplete instanceof MatAutocomplete) {
+      this._displayWithSubscription = this.autocomplete.displayWithChange
+        .subscribe(() => {
+          this.writeValue(this._value);
+        });
+    }
+  }
+  autocomplete: MatAutocomplete;
```

Comment:
Having the property like this means that people could still swap out the `autocomplete` programmatically without going through the setter.

Reply:
Okay, exposed autocomplete through a getter. I considered making autocomplete completely private and making a public autocompleteId property, but I was afraid that might break things.